### PR TITLE
feat: nixpkgsissue activity log view

### PR DESCRIPTION
**Pull request by alejandrosame**
feat: nixpkgsissue activity log view
    
Improve the messy representation of the different `pghistory` event tables:
    
- Create a view `log_nixpkgsissue` in the PostgreSQL backend and introduce an unmanaged model `NixpkgsIssueLogView` to display a list of changes per timestamp and user.
- Register `NixpkgsIssueLogView` in the `pghistory` app to declutter the log reporting from the already crowded `shared` section.
- Delete the previous activitiy log models and views in favour of the unified view model.
    
If performance starts being problematic for the `log_nixpkgsissue` view, we can
    make it a materialized view and add a trigger to update its contents on `shared.nixpkgsissue`
    and it's m2m relationships updates.

**NOTE:** This PR includes #101

### CODE REVIEW

File: src/website/shared/models/cve.py
Comment (RaitoBezarius) [inline src/website/shared/models/cve.py:307]:
```suggestion
# `cve` and `derivations` fields have to be tracked with via their `through` models
```

File: src/website/shared/models/cve.py
Comment (RaitoBezarius) [inline src/website/shared/models/cve.py:379]: Can we add a comment on why this is not managed by the ORM?

**Review by RaitoBezarius — APPROVED** [review]
Looks mostly good to me, I am just thinking about the GDPR considerations with the activity log, it's a good idea to make the cascade do nothing in case of deletion, but I wonder if we ought to implement a soft delete operation and bake in GDPR stuff in there directly.

File: src/website/shared/models/cve.py
Comment (alejandrosame) [reply src/website/shared/models/cve.py:379]: I pasted the wrong commit message in the PR description. I'll add similar comments in the model + the fact that Django doesn't seem to support multi table proxy models (which would be the equivalent of VIEW tables in ORM).

**Comment by alejandrosame** [comment]
@RaitoBezarius the cascade does nothing because it prevents Django from trying to delete content from a VIEW, which would just yield an error (it's read-only after all).

The data is collected from other `auth` and `pghistory` tables so we have to double check there what exactly we want to do on user deletion. There's also the default Django `admin` table, which registers less finegrained changes made by users.

I'll make an issue to not to forget about looking into this.

File: src/website/shared/models/cve.py
Comment (alejandrosame) [reply src/website/shared/models/cve.py]: fixed

File: src/website/shared/models/cve.py
Comment (alejandrosame) [reply src/website/shared/models/cve.py:382]: comment added

**Comment by alejandrosame** [comment]
Please don't merge yet. Working on some other functionality I noticed a mistake on the way I set the migration file in this PR.

I'll take this out of draft as soon as I push the fix.

**Comment by alejandrosame** [comment]
The PR is in a mergeable state again.

**Comment by alejandrosame** [comment]
rebased with latest changes

**Comment by alejandrosame** [comment]
@RaitoBezarius if you can reconfirm that you are ok with this approach after the rebases, this is ready to be merged.

**Comment by fricklerhandwerk** [comment]
Towards https://github.com/Nix-Security-WG/nix-security-tracker/issues/169

**Comment by alejandrosame** [comment]
We have activity log for suggestions with #366. If we were to reintroduce activity logs for issues, it should be based on top of what was implemented there.

**Pull request closed without merging**
